dbms/src/Columns/ColumnString.cpp

```diff
@@ -596,116 +919,168 @@ void ColumnString::serializeToPosForColumnArrayImpl(
         array_offsets.back(),
         size());
 
-    /// countSerializeByteSizeForColumnArray has already checked that the size of one element is not greater than UINT32_MAX
-    for (size_t i = 0; i < length; ++i)
+    /// countSerializeByteSizeForCmpColumnArray has already checked that the size of one element is not greater than UINT32_MAX
```

Comment:
ditto

Reply:
It's not easy to merge these two code path. Because the older code path will copy the whole row (including multiple string) into arena at once. But the new code path need to decode each string of this row.